pyo3-derive-backend/src/from_pyobject.rs

```diff
@@ -0,0 +1,545 @@
+                        }
+                    }
+                    other => {
+                        return Err(spanned_err(
+                            other,
+                            "Expected `transparent` or `annotation = \"name\"`",
+                        ))
+                    }
+                }
+            } else {
+                return Err(spanned_err(
+                    meta,
+                    "Unknown container attribute, expected `transparent` or \
+                    `annotation(\"err_name\")`",
+                ));
+            }
+        }
+        Ok(attrs)
+    }
+}
+
+/// Attributes for deriving FromPyObject scoped on fields.
+#[derive(Clone, Debug)]
+enum FieldAttribute {
+    GetItem(Option<syn::Lit>),
+    GetAttr(Option<syn::LitStr>),
+}
+
+impl FieldAttribute {
+    /// Extract the field attribute.
+    ///
+    /// Currently fails if more than 1 attribute is passed in `pyo3`
+    fn parse_attrs(attrs: &[Attribute]) -> Result<Option<Self>> {
+        let list = get_pyo3_meta_list(attrs)?;
+        if list.nested.is_empty() {
+            return Ok(None);
+        }
+        if list.nested.len() > 1 {
+            return Err(spanned_err(
+                list.nested,
+                "Only one of `item`, `attribute` can be provided, possibly with an \
+                additional argument: `item(\"key\")` or `attribute(\"name\").",
+            ));
+        }
+        let metaitem = list.nested.into_iter().next().unwrap();
+        let meta = match metaitem {
+            syn::NestedMeta::Meta(meta) => meta,
+            syn::NestedMeta::Lit(lit) => {
+                return Err(spanned_err(
+                    lit,
+                    "Expected `attribute` or `item`, not a literal.",
+                ))
+            }
+        };
+        let path = meta.path();
+        if path.is_ident("attribute") {
+            Ok(Some(FieldAttribute::GetAttr(Self::attribute_arg(meta)?)))
+        } else if path.is_ident("item") {
+            Ok(Some(FieldAttribute::GetItem(Self::item_arg(meta)?)))
+        } else {
+            Err(spanned_err(meta, "Expected `attribute` or `item`."))
+        }
+    }
+
+    fn attribute_arg(meta: syn::Meta) -> syn::Result<Option<syn::LitStr>> {
+        let arg_list = match meta {
+            syn::Meta::List(list) => list,
+            syn::Meta::Path(_) => return Ok(None),
+            Meta::NameValue(nv) => {
+                let err_msg = "Expected a string literal or no argument: `pyo3(attribute(\"name\") or `pyo3(attribute)`";
+                return Err(spanned_err(nv, err_msg));
+            }
+        };
+        let arg_msg = "Expected a single string literal argument.";
+        if arg_list.nested.is_empty() {
+            return Err(spanned_err(arg_list, arg_msg));
+        } else if arg_list.nested.len() > 1 {
+            return Err(spanned_err(arg_list.nested, arg_msg));
+        }
+        let first = arg_list.nested.first().unwrap();
```

Comment:
Don't `arg_list` and `arg_list.nested` have almost the same span?
```suggestion
        let first = match arg_list.nested.len() {
            1 => arg_list.nested.first().unwrap(),
            _ => return Err(spanned_err(arg_list, arg_msg)),
        };
```